platform/posix/transport/src/openssl_posix.c

```diff
@@ -223,22 +225,37 @@ static OpensslStatus_t convertToOpensslStatus( SocketStatus_t socketStatus )
 }
 /*-----------------------------------------------------------*/
 
-static OpensslStatus_t tlsHandshake( OpensslParams_t * pOpensslParams,
+static OpensslStatus_t tlsHandshake( const ServerInfo_t * pServerInfo,
+                                     OpensslParams_t * pOpensslParams,
                                      const OpensslCredentials_t * pOpensslCredentials )
 {
     OpensslStatus_t returnStatus = OPENSSL_SUCCESS;
     int32_t sslStatus = -1, verifyPeerCertStatus = X509_V_OK;
 
-    /* Enable SSL peer verification. */
-    SSL_set_verify( pOpensslParams->pSsl, SSL_VERIFY_PEER, NULL );
-
-    /* Setup the socket to use for communication. */
-    sslStatus = SSL_set_fd( pOpensslParams->pSsl, pOpensslParams->socketDescriptor );
+    /* Validate the hostname against the server's certificate. */
+    sslStatus = SSL_set1_host( pOpensslParams->pSsl,
+                               pServerInfo->pHostName );
 
     if( sslStatus != 1 )
     {
-        LogError( ( "SSL_set_fd failed to set the socket fd to SSL context." ) );
-        returnStatus = OPENSSL_API_ERROR;
+        LogError( ( "SSL_set1_host failed to validate the hostname "
+                    "in the server's certificate." ) );
```

Comment:
Are we actually validating the hostname here, or just setting the hostname for openssl to verify? It looks like the handshake happens below this

Reply:
Yes, it's just setting it to validate later.